frontend/src/utilities/__tests__/time.spec.ts

```diff
@@ -1,6 +1,205 @@
+    expect(relativeDuration(123456)).toBe('2:03');
+  });
+
+  it('should calculate values if minutes is less than 0', () => {
+    expect(relativeDuration(-123456)).toBe('0:0-124');
+  });
+});
+
+describe('convertDateToSimpleDateString', () => {
+  it('should convert date to simple date string', () => {
+    const date = new Date('2021-01-01');
+    expect(convertDateToSimpleDateString(date)).toBe('2021-01-01');
+  });
+
+  it('should convert date and time to simple string', () => {
+    const dateTime = new Date('2021-01-01T15:30:00');
+    expect(convertDateToSimpleDateString(dateTime)).toBe('2021-01-01');
+  });
+
+  it('should return null if the date is not equal', () => {
+    expect(convertDateToSimpleDateString()).toBe(null);
+  });
+});
+
+describe('convertDateToTimeString', () => {
+  it('should convert date to time string', () => {
+    const date = new Date('2021-01-01T15:30:00');
+    expect(convertDateToTimeString(date)).toBe('3:30 PM');
+  });
+
+  it('should return null if the date is not equal', () => {
+    expect(convertDateToTimeString()).toBe(null);
+  });
+
+  it('should convert to 12:00 AM if time entered on date is 0', () => {
+    const date = new Date('2021-01-01T00:00:00');
+    expect(convertDateToTimeString(date)).toBe('12:00 AM');
+  });
+});
+
+describe('ensureTimeFormat', () => {
+  it('should return time value', () => {
+    expect(ensureTimeFormat('3:30 PM')).toBe('3:30 PM');
+  });
+
+  it('should ensure time format', () => {
+    expect(ensureTimeFormat('')).toBe(null);
+  });
+
+  it('should return matching time format', () => {
+    expect(ensureTimeFormat('3:30PM')).toBe('3:30 PM');
+  });
+});
+
+describe('printSeconds', () => {
+  it('should print seconds', () => {
+    expect(printSeconds(3661)).toBe('1 second, 1 minute, 1 hour');
+  });
+
+  it('should handle a single unit', () => {
+    expect(printSeconds(5)).toBe('5 seconds');
+  });
+
+  it('should handle a single unit at max', () => {
+    expect(printSeconds(60)).toBe('1 minute');
+  });
+});
+
+describe('relativeTime', () => {
+  it('should convert milliseconds to minutes and seconds', () => {
+    expect(relativeTime(102, 30)).toBe('Just now');
+  });
+
+  it('should print ago if the current time is less than the previous', () => {
+    expect(relativeTime(62000, 1)).toBe('1 minute ago');
+  });
+
+  it('should print in if the elapsed time is less than 0', () => {
+    expect(relativeTime(1, 700000)).toBe('in 12 minutes');
+  });
+
+  it('should print in if the elapsed time is less than 0', () => {
+    expect(relativeTime(1, 62000)).toBe('in 1 minute');
+  });
+
+  it('should return just now when previous is NaN', () => {
+    expect(relativeTime(1000, NaN)).toBe('Just now');
+  });
+
+  it('should return just now if elapsed is less than 0', () => {
+    expect(relativeTime(1000, 2000)).toBe('Just now');
+  });
+
+  it('should return formatted time in minutes when elapsed time is less than an hour', () => {
+    const current = Date.now();
```

Comment:
In general I don't like the use of `Date.now()` in tests because the value isn't consistent. Even though these tests won't be affected by a changing value, I prefer that we avoid using `Date.now()` so it's not repeated in a case where it may have an affect on a test.